src/rpc/rpc.cc

```diff
@@ -1155,14 +1155,18 @@ future<> server::connection::send_unknown_verb_reply(std::optional<rpc_clock_typ
                           return send_unknown_verb_reply(timeout, msg_id, type);
                       }
 
+                      gate::holder guard;
+                      try {
+                          guard = h->use_gate.hold();
+                      } catch (const gate_closed_exception&) {
+                          return send_unknown_verb_reply(timeout, msg_id, type);
```

Comment:
I would have made `get_handler` return `std::pair<rpc_handler*, gate::holder>` to not repeat error handling here, but this is fine as well.

Reply:
ok, I'll update. I also forgot to add a test to `rpc_test.cc`